pbsmetrics/go_metrics.go

```diff
@@ -562,6 +582,14 @@ func (me *Metrics) RecordTimeoutNotice(success bool) {
 	return
 }
 
+func (me *Metrics) RecordTCFReq(version int) {
+	if version >= 0 && version < len(me.TCFReqVersion) {
+		me.TCFReqVersion[version].Mark(1)
```

Comment:
The use of index to correlate to a version string relies on the specific order of `TCFVersions()`. Do other metrics really do this? Would a map by the version string be safer?

Reply:
Other metrics do not do this. They resolve the value upstream, and pass the code to the Record method. In this case, we already have version resolved to an integer, and felt it would make more sense to stringify it on the metrics side, which is probably the first place we will look if at some point down the line we want to change the strings that get recorded.

Reply:
If I were to do this the way other metrics do it, I would have some code that looks like:
`if version == 1 {
    val = TCFVersionV1
} else if version == 2{
    val = TCFVersionV2
} else {
    val = TCFVersionErr
}`
However all the other metrics already produce lists of values in a predefined order. They just don't have an integer predefined for the value they represent that makes a good index. It is usually decoding HTTP status codes or checking error types, or just hardcoded based on the endpoint code it is in. Version number is unique in this way that it is easier to use the version number as the index to the version number than to have a set of ifs or a case statement to resolve the the version number to the stringify the version number. Or go back to my original version where I do not predefine the version strings, but generate them on the fly with `Sprinft()`